api/groups/urlParams.go

```diff
@@ -84,18 +114,52 @@ func parseStringUrlParam(c *gin.Context, name string) string {
 	return c.Request.URL.Query().Get(name)
 }
 
-func parseUintUrlParam(c *gin.Context, name string) (uint32, error) {
+func parseUint32UrlParam(c *gin.Context, name string) (core.OptionalUint32, error) {
 	param := c.Request.URL.Query().Get(name)
 	if param == "" {
-		return 0, nil
+		return core.OptionalUint32{}, nil
 	}
 
 	value, err := strconv.ParseUint(param, 10, 32)
 	if err != nil {
-		return 0, err
+		return core.OptionalUint32{}, err
+	}
+
+	return core.OptionalUint32{
+		Value:    uint32(value),
+		HasValue: true,
+	}, nil
+}
+
+func parseUint64UrlParam(c *gin.Context, name string) (core.OptionalUint64, error) {
```

Comment:
It should be mentioned that we also have these utility functions for parsing here: 

https://github.com/ElrondNetwork/elrond-go/blob/master/api/groups/urlParams.go

Moving them to a shared library such as `elrond-go-gin-helpers` would be a bit too soon at this moment, perhaps.

What do you think?

Reply:
We might create a repo to hold all wrappers we use for creating a web server, a web client and so on as to remove duplicated code across repos.
It can be named `elrond-go-api` or something like this

Reply:
agree, was something I suggested some time ago as well.